tklinenums/tklinenums.py

```diff
@@ -48,45 +54,69 @@ def __init__(
             *args,
             **kwargs,
         )
-    
-        # Set variable
-        self.textwidget = editor
+
+        # Set variables
+        self.textwidget = textwidget
         self.master = master
         self.justify = justify
         self.click_pos: None = None
+        self.colors = colors
 
-        # Set style
-        self.set_to_ttk_style()
+        # Set style and its binding
+        self.set_colors()
+        self.bind("<<ThemeChanged>>", self.set_colors, add=True)
 
-        # Event bindings
+        # Mouse scroll binding
         self.bind("<MouseWheel>", self.mouse_scroll, add=True)
+        self.bind("<Button-4>", self.mouse_scroll, add=True)
+        self.bind("<Button-5>", self.mouse_scroll, add=True)
+
+        # Click bindings
         self.bind("<Button-1>", self.click_see, add=True)
         self.bind("<ButtonRelease-1>", self.unclick, add=True)
         self.bind("<Double-Button-1>", self.double_click, add=True)
-        self.bind("<Button1-Motion>", self.drag, add=True)
-        self.bind("<Button1-Leave>", self.auto_scroll, add=True)
+
+        # Mouse drag bindings
+        self.bind("<Button1-Motion>", self.in_widget_select_mouse_drag, add=True)
+        self.bind("<Button1-Leave>", self.mouse_off_screen_scroll, add=True)
+        self.bind("<Button1-Enter>", self.stop_mouse_off_screen_scroll, add=True)
 
         # Set the yscrollcommand of the text widget to redraw the widget
-        editor["yscrollcommand"] = self.redraw
+        textwidget["yscrollcommand"] = self.redraw
+
+        # Redraw the widget
+        self.redraw()
 
     def redraw(self, *args) -> None:
         """Redraws the widget"""
-        # Resize the widget based on the number of lines in the editor
+
+        # Resize the widget based on the number of lines in the textwidget and set colors
+        del args
```

Comment:
This strikes me as odd... why do this?

Reply:
It provides no purpose and it's better to show that it has no purpose by deleting the local variable immediately than leaving people confused what it's for.

Reply:
Then you should name the variable `_`

Reply:
Will do.

Reply:
```
def redraw(*_, **__):
```
ftw ;)

to be honest, I don't normally worry about having unused things defined in the function signature.  I don't bother with `_`, since ruff/flake8/etc don't complain about unused variables.  This is especially true when using frameworks/callbacks.

If I need to pass things to `super()` that _I_ don't care about, I do this:
```python
def some_override(self, myvar: str, *args, **kwargs):
    self.myvar = myvar
    super().__init__(*args, **kwargs)
```

as always, ymmv